stdlib/Dates/src/rounding.jl

```diff
@@ -114,147 +114,147 @@ function Base.floor(x::ConvertiblePeriod, precision::T) where T <: ConvertiblePe
 end
 
 """
-    floor(dt::TimeType, p::Period) -> TimeType
+    floor(p::Period, dt::TimeType) -> TimeType
 
 Return the nearest `Date` or `DateTime` less than or equal to `dt` at resolution `p`.
 
-For convenience, `p` may be a type instead of a value: `floor(dt, Dates.Hour)` is a shortcut
-for `floor(dt, Dates.Hour(1))`.
+For convenience, `p` may be a type instead of a value: `floor(Dates.Hour, dt)` is a shortcut
+for `floor(Dates.Hour(1), dt)`.
 
 ```jldoctest
-julia> floor(Date(1985, 8, 16), Month)
+julia> floor(Month, Date(1985, 8, 16))
 1985-08-01
 
-julia> floor(DateTime(2013, 2, 13, 0, 31, 20), Minute(15))
+julia> floor(Minute(15), DateTime(2013, 2, 13, 0, 31, 20))
 2013-02-13T00:30:00
 
-julia> floor(DateTime(2016, 8, 6, 12, 0, 0), Day)
+julia> floor(Day, DateTime(2016, 8, 6, 12, 0, 0))
 2016-08-06T00:00:00
 ```
 """
-Base.floor(::Dates.TimeType, ::Dates.Period)
+Base.floor(::Dates.Period, ::Dates.TimeType)
 
 """
     ceil(dt::TimeType, p::Period) -> TimeType
 
 Return the nearest `Date` or `DateTime` greater than or equal to `dt` at resolution `p`.
 
-For convenience, `p` may be a type instead of a value: `ceil(dt, Dates.Hour)` is a shortcut
-for `ceil(dt, Dates.Hour(1))`.
+For convenience, `p` may be a type instead of a value: `ceil(Dates.Hour, dt)` is a shortcut
+for `ceil(Dates.Hour(1), dt)`.
 
 ```jldoctest
-julia> ceil(Date(1985, 8, 16), Month)
+julia> ceil(Month, Date(1985, 8, 16))
 1985-09-01
 
-julia> ceil(DateTime(2013, 2, 13, 0, 31, 20), Minute(15))
+julia> ceil(Minute(15), DateTime(2013, 2, 13, 0, 31, 20))
 2013-02-13T00:45:00
 
-julia> ceil(DateTime(2016, 8, 6, 12, 0, 0), Day)
+julia> ceil(Day, DateTime(2016, 8, 6, 12, 0, 0))
 2016-08-07T00:00:00
 ```
 """
-function Base.ceil(dt::TimeType, p::Period)
-    f = floor(dt, p)
+function Base.ceil(p::Period, dt::TimeType)
+    f = floor(p, dt)
     return (dt == f) ? f : f + p
 end
 
 """
-    ceil(x::Period, precision::T) where T <: Union{TimePeriod, Week, Day} -> T
+    ceil(precision::T, x::Period) where T <: Union{TimePeriod, Week, Day} -> T
 
 Round `x` up to the nearest multiple of `precision`. If `x` and `precision` are different
 subtypes of `Period`, the return value will have the same type as `precision`.
 
-For convenience, `precision` may be a type instead of a value: `ceil(x, Dates.Hour)` is a
-shortcut for `ceil(x, Dates.Hour(1))`.
+For convenience, `precision` may be a type instead of a value: `ceil(Dates.Hour, x)` is a
+shortcut for `ceil(Dates.Hour(1), x)`.
```

Comment:
This statement recommends the use of deprecated functions. It should probably be replaced with a warning noting that the argument order of this `ceil` method differs from the standard rounding argument order.

Reply:
Its true for floor, ceil and round. So, I replaced all of those lines with warnings.

Reply:
@LilithHafner Added `!!! warning`. Its also returning proper values this time. Can you review the code?